File: lei_lds2122_grupo2-main/Backend/Data/ViewModels/Request/GenerateRouteVM.cs
namespace Backend.Data.ViewModels.Request
{
    /// <summary>
    /// View Model class for Creat User reqeust
    /// </summary>
    public class GenerateRouteVM
    {
        public string Name { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime FinishTime { get; set; }
        public Boolean VisitedPlaces { get; set; }
        public List<string> Categories { get; set; }

        /// <summary>
        /// Class' constructor method
        /// </summary>
        /// <param name="name">Route's name</param>
        /// <param name="startTime">Time that route starts</param>
        /// <param name="finishTime">Time that route ends</param>
        /// <param name="visitedPlaces">Whether or not the user wants to visit places already visited</param>
        /// <param name="categories">Categories of palces that user chose</param>
        public GenerateRouteVM(string name, DateTime startTime, DateTime finishTime, Boolean visitedPlaces, 
        List<String> categories)
        {
            this.Name = name;
            this.StartTime = startTime;
            this.FinishTime = finishTime;
            this.VisitedPlaces = visitedPlaces;
            this.Categories = categories;
        }
    }
}

File: lei_lds2122_grupo2-main/Frontend/src/app/viewModels/requests/GenerateRouteVM.ts
import { toJSONString } from "../interfaces/toJSONString";

/**
 * View Model class for a Create User request
 */
export class GenerateRouteVM implements toJSONString {

    /**
     * Construct's an instance of {@link GenerateRouteVM}
     * 
     * @param name route's name.
     * @param startTime route's start date.
     * @param finishTime route's finish date.
     * @param visitedPlaces Specifies if the generated route should have palces already visited.
     * @param categories route's categories.
     */
    constructor(
        public name: string,
        public startTime: Date,
        public finishTime: Date,
        public visitedPlaces: Boolean,
        public categories: Array<string>
    ) { }

    /**
     * Converts the class into a string representation of a json object 
     * 
     * @returns string representation of the object
     */
    toJSONString(): string {
        return JSON.stringify(this)
    }

}